src/main/resources/static/admin.js

```diff
@@ -0,0 +1,58 @@
+
+$("#crawl-form").submit(function (e) {
+
+    e.preventDefault()
+
+    let enteredURL = $("#url-search").val().trim();
+
+    if (enteredURL.length === 0) {
+        alert("Empty string detected. Please use enter a valid URL.");
+        return;
+    }
+
+    console.log("URL: " + enteredURL)
+
+    document.getElementById("crawl-response").innerHTML = "";
+
+    let adminFirstName = localStorage.getItem("adminFirstName")
+    let adminLastName = localStorage.getItem("adminLastName")
+    let adminUsername = localStorage.getItem("adminUsername")
+
+    if (adminFirstName == null || adminLastName == null || adminUsername == null) {
+        alert("Please login before trying to perform an administrative task. Redirecting to login ... ")
+        window.location.href = window.location.origin + "/login.html";
+        return;
+    }
+
+    const crawlURL = window.location.origin + "/api/admin/" + adminUsername + "/crawl?url=" + enteredURL
+
+    const xhrCrawl = new XMLHttpRequest();
+
+    xhrCrawl.open('GET', crawlURL);
+    xhrCrawl.responseType = 'text';
+
+    xhrCrawl.onload = function(e) {
```

Comment:
Is this the same `e` as on lines 2 and 4? And if so, is this a naming convention that I am not aware of (as a non-JS programmer), or could it be changed to a more descriptive name?

Reply:
That e is referred there is known as event. submit takes an event on submit. Could have been named as event but I followed an internet resource and have forgotten to rename a bit better.

Reply:
Ah, makes sense. I think going with the convention is fine here, but as a non-front-end dev, I'm used to `e` being `exception`.